sdk/storage/azure-storage-file-share/azure/storage/fileshare/_share_client.py

```diff
@@ -415,7 +424,14 @@ def delete_share(
         timeout = kwargs.pop('timeout', None)
         delete_include = None
         if delete_snapshots:
+            warn("The `delete_snapshots` parameter is being deprecated. "
+                 "Please use the `include_snapshots` parameter instead.", DeprecationWarning)
             delete_include = DeleteSnapshotsOptionType.include
+        if include_snapshots:
+            if isinstance(include_snapshots, str) and include_snapshots not in ['include', 'include-leased']:
```

Comment:
in .NET and Java we call it `include` and `include-with-leased` as we find original naming in rest api confusing.
I.e. it's not 100% clear if `include-leased` means include only leased or everything with leased.

Reply:
I agree, the naming of this attribute is quite.... misleading. I'll create a wrapper to rename this in Python as well.